src/os/bluestore/BlueStore.cc

```diff
@@ -6322,11 +6323,19 @@ void BlueStore::_txc_state_proc(TransContext *txc)
 		   << dendl;
 	} else {
 	  _txc_finalize_kv(txc, txc->t);
+	  if (g_conf->bluestore_rtc && txc->wal_txn && !txc->no_rtc) {
+	    if (txc->osr->kv_finisher_submitting == 1) {
+	      db->submit_transaction_sync(txc->t);
+	      _txc_state_proc(txc);
+	      return;
```

Comment:
If I'm understanding correctly, the goal here is to avoid the context switch into the kv_sync_thread when the kv commits are blazing fast (e.g., nvdimm).  That seems reasonable.  What does that have to do with wal_txn, though?  Wouldn't we want to do this regardless of whether there is wal IO?  (There usually isn't!).